agent/app/agent_unix.go

```diff
@@ -48,6 +48,7 @@ const initPID = 1
 var awsVPCCNIPlugins = []string{ecscni.ECSENIPluginName,
 	ecscni.ECSBridgePluginName,
 	ecscni.ECSIPAMPluginName,
+	ecscni.ECSAppMeshPluginName,
```

Comment:
`ECSAppMeshPluginName` shouldn't be a part of this list. It's not needed for the `awsvpc` mode. However, what you do need is a way to check if the plugin binary file exists or not. Here's an example: https://github.com/aws/amazon-ecs-agent/blob/9708e515dd932f7b8d9a0a1be9046c82b92d89a0/agent/app/agent_capability.go#L249

Reply:
Good point. I actually tried a bit and i noticed our currently testing logic in agent capabilities needs quite effort to refactor to make it work for both linux and windows. Discussed offline, I created a issue https://github.com/aws/amazon-ecs-agent/issues/1893 to work as follow up